pkgs/development/libraries/qt-6/modules/qtdeclarative.nix

```diff
@@ -29,11 +39,160 @@ qtModule {
   ];
 
   # debug: install is failing
-  splitBuildInstall = {
+  splitBuildInstall =
+  let
+    # workaround for splitBuildInstall
+    pname = "qtdeclarative";
+    version = "6.2.1";
+    self = { inherit qtbase; };
+    args = { postFixup = ""; };
+  in
+  {
     # needed to disable rebuild
     # TODO simpler way? cmake hooks trigger rebuild?
+    /*
     installPhase = ''
       cmake -P cmake_install.cmake
     '';
+    */
+
+
+  # copy from nixpkgs/pkgs/development/libraries/qt-6/qtModule.nix
+  # TODO move back when working
+  postFixup = ''
+    if [ -d "''${!outputDev}/lib/pkgconfig" ]; then
+        find "''${!outputDev}/lib/pkgconfig" -name '*.pc' | while read pc; do
+            sed -i "$pc" \
+                -e "/^prefix=/ c prefix=''${!outputLib}" \
+                -e "/^exec_prefix=/ c exec_prefix=''${!outputBin}" \
+                -e "/^includedir=/ c includedir=''${!outputDev}/include"
+        done
+    fi
+
+    # TODO refactor. same code in qtbase.nix and qtModule.nix
+    echo "patching output paths in cmake files ..."
+    (
+    cd $dev/lib/cmake
+    moduleNAME="${lib.toUpper pname}"
+    outEscaped=$(echo $out | sed 's,/,\\/,g')
+    devEscaped=$(echo $dev | sed 's,/,\\/,g')
+    if [ -n "$bin" ]; then
+    binEscaped=$(echo $bin | sed 's,/,\\/,g') # optional plugins
+    else binEscaped=""; fi
+
+    # TODO build the perlRegex string with nix? avoid the bash escape hell
+    # or use: read -d "" perlRegex <<EOF ... EOF
+    s=""
+    s+="s/^# Compute the installation prefix relative to this file\."
+    s+="\n.*?set\(_IMPORT_PREFIX \"\"\)\nendif\(\)"
+    s+="/# NixOS was here"
+    s+="\nset(_''${moduleNAME}_NIX_OUT \"$outEscaped\")"
+    s+="\nset(_''${moduleNAME}_NIX_DEV \"$devEscaped\")"
+    s+="\nset(_''${moduleNAME}_NIX_BIN \"$binEscaped\")/s;"
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?include/\\\''${_''${moduleNAME}_NIX_DEV}\/include/g;"
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?libexec/\\\''${_''${moduleNAME}_NIX_OUT}\/libexec/g;"
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?lib/\\\''${_''${moduleNAME}_NIX_OUT}\/lib/g;" # must come after libexec
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?plugins/\\\''${_''${moduleNAME}_NIX_BIN}\/lib\/qt-${version}\/plugins/g;"
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?bin/\\\''${_''${moduleNAME}_NIX_DEV}\/bin/g;" # qmake ...
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?mkspecs/\\\''${_''${moduleNAME}_NIX_DEV}\/mkspecs/g;"
+    s+="s/\\\''${_IMPORT_PREFIX}\/(\.\/)?qml/\\\''${_''${moduleNAME}_NIX_OUT}\/lib\/qt-${version}\/lib\/qml/g;"
```

Comment:
by patching the `qml` paths, the `qtdeclarative` build is passing
but `libQtQuick.so` is not usable from cmake projects

for example [qtquicktimeline](/home/user/src/nixos/nixpkgs-2021-10-12/nixpkgs/pkgs/development/libraries/qt-6/modules/qtquicktimeline.nix) requires QtQuick

```cmake
find_package(Qt6 6.2.1 QUIET CONFIG OPTIONAL_COMPONENTS Quick)

if(NOT TARGET Qt::Quick)
    message(NOTICE "Skipping the build as the condition \"TARGET Qt::Quick\" is not met.")
    return()
endif()

# -> Skipping the build as the condition "TARGET Qt::Quick" is not met.
```

per `cmakeFlags = [ "-DCMAKE_FIND_DEBUG_MODE=TRUE" ];`
this will use `qtbase-6.2.1-dev/lib/cmake/Qt6/Qt6Config.cmake`
but `Qt6Config.cmake` cannot find `qtdeclarative`

`CMAKE_LIBRARY_PATH` has `/nix/store/qhh0spm39dvi6xkg386721h3gsb4p9an-qtdeclarative-6.2.1/lib`

but we would need something like
`CMAKE_CONFIG_PATH` with `/nix/store/gslwy5b5pfvjdi8p43q2hklg6nqafjg4-qtdeclarative-6.2.1-dev/lib/cmake`

Reply:
next error:

```
  The imported target "Qt6::qmldom" references the file

     "/nix/store/gslwy5b5pfvjdi8p43q2hklg6nqafjg4-qtdeclarative-6.2.1-dev/bin/qmldom"

  but this file does not exist.  Possible reasons include:
```

... but the file is in `/nix/store/2s3nk9frssyib2qp29hp30i2i1500mp2-qtdeclarative-6.2.1/bin/qmldom`

&rarr; bug in `moveQtDevTools`?

&rarr; workaround in `qtdeclarative.nix`

```sh
mv $out/bin $dev/bin
```